torch/nn/functional.py

```diff
@@ -5404,7 +5404,7 @@ def pad(
         return handle_torch_function(
             torch.nn.functional.pad, (input,), input, pad, mode=mode, value=value
         )
-    if not torch.jit.is_scripting():
+    if not torch.jit.is_scripting() and not torch.compiler.is_compiling():
```

Comment:
i am not sure this is the right fix since none of the other functions in this file require this condition. why do we need it here but nowhere else?

Reply:
Right and there are lots of other functions in the file that have the same pattern? Why is it not an issue there?